pypki3/config.py

```diff
@@ -193,12 +193,24 @@ def loaded_encoded_pem(key_obj: Any, cert_obj: Any) -> LoadedPKIBytes:
     return LoadedPKIBytes(key_bytes, cert_bytes)
 
 def separate_pem(pem_data: bytes) -> Tuple[bytes, List[bytes]]:
-    certs = parse_pem(pem_data)
-
-    if len(certs) < 2:
-        raise Pypki3Exception('PEM must contain the key then certificate(s)')
-
-    return certs[0].as_bytes(), [x.as_bytes() for x in certs[1:]]
+    pem_objs = parse_pem(pem_data)
+    key = None
+    certs = list()
+    
+    for pem_obj in reversed(pem_objs):
+        if isinstance(pem_obj, PrivateKey):
+            key = pem_obj
+        elif isinstance(pem_obj, Certificate):
+            certs.append(pem_obj)
+
+    if not key and not certs:
+        raise Pypki3Exception('PEM missing key and certificate(s)')
+    elif key and not certs:
+        raise Pypki3Exception('PEM contains key but is missing certificate(s)')
+    elif certs and not key:
```

Comment:
This `elif` can just be `if` since it comes after `raise`.

Reply:
Cleaned up in cbe026c.